packages/test/test-utils/src/containerRuntimeFactories.ts

```diff
@@ -0,0 +1,198 @@
+
+	/**
+	 * The runtime options passed to the IContainerRuntime when instantiating it
+	 */
+	readonly runtimeOptions?: IContainerRuntimeOptions;
+
+	/**
+	 * Function that will initialize the entryPoint of the IContainerRuntime instances
+	 * created with this factory
+	 */
+	readonly provideEntryPoint?: (runtime: IContainerRuntime) => Promise<FluidObject>;
+}
+
+/**
+ * A ContainerRuntimeFactory that initializes Containers with a single default data store, which can be requested from
+ * the container with an empty URL.
+ *
+ * @deprecated
+ * Do not reference this type directly. It will be removed in the future.
+ * E.g. use {@link IRuntimeFactory} instead.
+ */
+export class ContainerRuntimeFactoryWithDefaultDataStore
+	extends RuntimeFactoryHelper
+	implements IProvideFluidDataStoreRegistry
+{
+	public static readonly defaultDataStoreId = defaultDataStoreId;
+
+	/**
+	 * {@inheritDoc @fluidframework/runtime-definitions#IProvideFluidDataStoreRegistry.IFluidDataStoreRegistry}
+	 */
+	public get IFluidDataStoreRegistry(): IFluidDataStoreRegistry {
+		return this.registry;
+	}
+
+	protected readonly defaultFactory: IFluidDataStoreFactory;
+
+	private readonly registry: IFluidDataStoreRegistry;
+
+	/**
+	 * {@inheritDoc ContainerRuntimeFactoryWithDefaultDataStoreProps.registryEntries}
+	 */
+	private readonly registryEntries: NamedFluidDataStoreRegistryEntries;
+
+	/**
+	 * {@inheritDoc ContainerRuntimeFactoryWithDefaultDataStoreProps.runtimeOptions}
+	 */
+	private readonly runtimeOptions?: IContainerRuntimeOptions;
+
+	// eslint-disable-next-line import/no-deprecated
+	private readonly requestHandlers: readonly RuntimeRequestHandler[];
+
+	/**
+	 * {@inheritDoc ContainerRuntimeFactoryWithDefaultDataStoreProps.provideEntryPoint}
+	 */
+	private readonly provideEntryPoint: (runtime: IContainerRuntime) => Promise<FluidObject>;
+
+	public constructor(props: ContainerRuntimeFactoryWithDefaultDataStoreProps) {
+		super();
+
+		const getDefaultObject = async (
+			request: IRequest,
+			runtime: IContainerRuntime,
+		): Promise<IResponse | undefined> => {
+			const parser = RequestParser.create(request);
+			if (parser.pathParts.length === 0) {
+				// This cast is safe as loadContainerRuntime is called in the base class
+				// eslint-disable-next-line import/no-deprecated
+				return (runtime as IContainerRuntimeWithResolveHandle_Deprecated).resolveHandle({
+					url: `/${defaultDataStoreId}${parser.query}`,
+					headers: request.headers,
+				});
+			}
+			return undefined; // continue search
+		};
+
+		this.defaultFactory = props.defaultFactory;
+		this.registryEntries = props.registryEntries;
+		this.runtimeOptions = props.runtimeOptions;
+		this.provideEntryPoint = props.provideEntryPoint ?? getDefaultFluidObject;
+		this.requestHandlers = [getDefaultObject];
+		this.registry = new FluidDataStoreRegistry(this.registryEntries);
+	}
+
+	/**
+	 * Called the one time the container is created, and not on any subsequent load.
```

Comment:
Why have a method that documents when its called? Either it should be a method someone can call, or implementing some interface from which it inherits the documentation. Documenting when a concreate method is called seems really odd. DO we expect subclasses to override this or something?

Reply:
Same for the methods below.

Maybe this should document when the user is supposed to call it now when it is called? That might make more sense if this is something we expect the user of this type to invoke?

Reply:
We have containerHasInitialized for overriding, so maybe this should be sealed?

Reply:
These were copied over from the original code. But these also unnecessarily deduplicate the docs on the base class, so I've just removed them. The way these types are documented generally are very poor.